File: filestorage-sample/src/main/java/ru/surfstudio/android/filestorage/sample/ui/base/configurator/ActivityConfigurator.java
package ru.surfstudio.android.filestorage.sample.ui.base.configurator;

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator;
import ru.surfstudio.android.filestorage.sample.app.App;
import ru.surfstudio.android.filestorage.sample.app.dagger.AppComponent;
import ru.surfstudio.android.filestorage.sample.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.filestorage.sample.ui.base.dagger.activity.ActivityModule;
import ru.surfstudio.android.filestorage.sample.ui.base.dagger.activity.DaggerActivityComponent;

/**
 * Базовый конфигуратор для активити
 */

public class ActivityConfigurator extends BaseActivityConfigurator<ActivityComponent, AppComponent> {

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(new ActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) getTargetActivity().getApplicationContext()).getAppComponent();
    }
}

File: navigation/sample-standard/src/main/java/ru/surfstudio/android/navigation/sample_standard/di/ui/configurator/ActivityConfigurator.kt
package ru.surfstudio.android.navigation.sample_standard.di.ui.configurator

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator
import ru.surfstudio.android.navigation.sample_standard.App
import ru.surfstudio.android.navigation.sample_standard.di.AppComponent
import ru.surfstudio.android.navigation.sample_standard.di.ui.ActivityComponent
import ru.surfstudio.android.navigation.sample_standard.di.ui.ActivityNavigationModule
import ru.surfstudio.android.navigation.sample_standard.di.ui.DaggerActivityComponent
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule

class ActivityConfigurator : BaseActivityConfigurator<ActivityComponent, AppComponent>() {

    @Override
    override fun createActivityComponent(parentComponent: AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule((DefaultActivityModule(persistentScope)))
                .activityNavigationModule(ActivityNavigationModule())
                .build()
    }

    @Override
    override fun getParentComponent(): AppComponent {
        val app = persistentScope.screenState.activity.applicationContext as App
        return app.appComponent
    }
}
